### ISSUE TITLE
Need to return and allow user input of lists

### ISSUE THREAD

**Issue by edeutsch** [RTXteam/RTX#242]
This was first suggested by @jaredroach , but has since become crystal clear that we really need a great way of getting a list out of each query. Right now we treat the response as a set of individual independent results or answers. Although the results are sort of list-ish, the interface is not amenable to a list. We need to return an easily usable list to the user.

One option is for the UI to parse the current JSON response and try to formulate a table. But that would require a lot of logic to understand things well enough to do that well. I might be able try this.

BUT, I thought it might be better to push this onto the reasoner. I suggest that the reasoner already has the knowledge to figure out how best to format the data, rather than having separate code try to figure it out by studying the data.  But this is debatable.

@dkoslicki I added several new elements to the schema in the last round. If you go here:
https://rtx.ncats.io/api/rtx/v1/ui/#!/query/query
and click on the gray [Model] button below "successful operation" you can see the model. Look in Result. You see this:
**essence (string, optional)**: A single string that is the terse essence of the result (useful for simple answers) ,
==> If the query is "which proteins are targets of ibuprofen?", then the essence should be each protein symbol. So PTGS1 and then PTGS2. This should be 30 characters or less. it is designed to displayed on all the collapsed box eaders so one can more easily see what the answers are.

**row_data (Array[string], optional)**: An arbitrary list of values that captures the main data of the result that can be turned into a tabular result across all answers (each result is a row) for a user that wants tabular output.
==> here you can put in any array you want, ideally faithfully displayed by the UI. So in the above example, it could be: [ "ibuprofen","PTGS1","0.96","P02534","CHEMBL.TARGET:CHEMBL2323" ] or whatever information the reasoner has on hand that it thinks the user might like to have in a table. Just put it in each result. Then, @isbluis could display it as a tabular widget in the response. maybe above all the individual answers with graphs and drill-downs. or maybe at the bottom, I don't know. but in an easy copy-paste format for getting into Excel or similar.

What do you think? We can alter this. This was just a snap decision back at the hackathon when I was making other major changes to the schema.

**Comment by jaredroach** [comment]
It seems that all the teams should agree on a format(s)/protocol for passing results. So that the result of one team can be given as input to another team. So if there is a universal agreed JSON format, then it would make some sense to write a parser from this format for purposes of UI and  for "list" input into RTX questions that take lists as input.

But the teams could also agree on a flat file 2-d annotated list format, for this purpose.

In short, I think I am just restating what @edeutsch wrote above.

I don't have much of a preference. It is not that hard to parse JSON. But computational biologists usually find it easier to work with "EXCEL"ish data.

**Comment by edeutsch** [comment]
exactly! Thus far it's a complex JSON format for the output. For input, we still need to figure out what we want it to look like. I suggest we pick an example and mock up how we'd like it to look. Once we have something we like, we can suggest to the other reasoners and iterate.

**Comment by dkoslicki** [comment]
@edeutsch I've added the ``essence``, ``row_data``, and ``table_column_names`` fields to all of the following question templates: Q3-Q12. For example, you now get this in the output JSON to the question "What are the protein targets of ibuprofen." For the row data, I'm only including the relevant bioentities: ``row_data = [source_name, source_id, target_name, target_id]``. If needed, I can add more stuff in (like confidences, scores, etc. but since those are really of limited use for down-stream analysis at this point, I though including just names and identifiers is sufficient for now as @isbluis gets this implemented with the UI).

**Comment by dkoslicki** [comment]
@edeutsch I've also added these attributes for the Similarity question templates (such as "what drugs have similar protein targets to ibuprofen"). Let me know if this is looking good and I can implement them for the other question templates as well.

**Comment by edeutsch** [comment]
Great, thanks! Luis and I need to work past one imminent deadline and then get to all this.

**Comment by saramsey** [comment]
@isbluis can we get an update on this, Luis?  Thank you!

**Comment by dkoslicki** [comment]
Closing due to #788, lists currently being returned in the UI (or accessible via ReasonerStd API response and #720)
